src/core.jl

```diff
@@ -0,0 +1,70 @@
+function reproject(input_data, output_projection; shape_out = nothing, order = 1, hdu_in = 1, hdu_out = 1, return_footprint = true)
+    if input_data isa ImageHDU
+        array_in, wcs_out = parse_input_data(input_data)
+    else
+        array_in, wcs_out = parse_input_data(input_data, hdu_in)
+    end
+    
+    if output_projection isa FITS || output_projection isa String
+        wcs_in, shape_out = parse_output_projection(output_projection, hdu_out)
+    else
+        wcs_in, shape_out = parse_output_projection(output_projection, shape_out)
+    end
+    
+    type_in = wcs_to_celestial_frame(wcs_in)
+    type_out = wcs_to_celestial_frame(wcs_out)
+    
+    if type_in == type_out
+        return array_in
+    end
+    
+    img_out = fill!(zeros(shape_out), NaN)
+    
+    if order == 0
+        itp = interpolate(array_in, BSpline(Constant()))
+    elseif order == 1
+        itp = interpolate(array_in, BSpline(Linear()))
+    else
+        itp = interpolate(array_in, BSpline(Quadratic(InPlace(OnCell()))))
+    end
+    
+    for i in 1:shape_out[1]
+        for j in 1:shape_out[2]
+            pix_cord_in = [float(i); float(j)]
```

Comment:
Is this supposed to be called `pix_coord_in`?  Also, does it need to be a vector or it can be a tuple as well?

Reply:
`pix_to_world` expects it to be a vector. Also, the name is intuitively correct as it it the pixel coordinates(from input image) which is getting converted to desired output image.

Reply:
I was referring to the fact that probably "cord" stands for "coordinates", with two "o"s rather than one.